traitlets/traitlets.py

```diff
@@ -2479,7 +2479,16 @@ def instance_init(self, obj):
 
     def from_string(self, s):
         """Load value from a single string"""
-        return self.from_string_list([s])
+        if not isinstance(s, str):
+            raise TraitError(f"Expected string, got {s!r}")
+        try:
+            test = literal_eval(s)
+        except Exception:
+            test = None
+        if isinstance(test, list):
+            return test
+        else:
+            raise TraitError(f"Expected to deserialise a list, got {s!r}")
```

Comment:
Are we expecting a list in general? Maybe we should be passing it to `self.validate`?

Reply:
I think you are right.